src/swap/redux/middleware.js

```diff
@@ -5,23 +5,53 @@ import {
 } from '../../utils/redux/templates/ethersTransactions'
 import { getAllBalancesForConnectedAddress } from '../../deltaBalances/redux/actions'
 import * as Swap from '../index'
-import { getOrderId } from '../../utils/order'
+import { getSwapSimpleOrderId } from '../../utils/order'
 
-async function swap(store, action) {
+async function fillSwapSimple(store, action) {
   const signer = await store.dispatch(getSigner())
   const { order } = action
-  return Swap.swap(order, signer)
+  return Swap.swapSimple(order, signer)
+}
+
+async function cancelSwap(store, action) {
+  const signer = await store.dispatch(getSigner())
+  const { order } = action
+  return Swap.cancelSwap(order, signer)
+}
+
+async function signSwapSimple(store, action) {
+  const signer = await store.dispatch(getSigner())
+
+  Swap.signSwapSimple(action, signer)
+    .then(order => {
+      action.resolve(order)
+    })
+    .catch(err => {
+      action.reject(err)
+    })
 }
 
 export default function walletMiddleware(store) {
   return next => action => {
     switch (action.type) {
-      case 'SWAP':
-        makeMiddlewareEthersTransactionsFn(swap, 'swap', store, action, getOrderId(action.order))
+      case 'FILL_SWAP_SIMPLE':
+        makeMiddlewareEthersTransactionsFn(
+          fillSwapSimple,
+          'fillSwapSimple',
+          store,
+          action,
+          getSwapSimpleOrderId(action.order),
+        )
         break
-      case makeEthersTxnsActionTypes('swap').mined:
+      case makeEthersTxnsActionTypes('fillSwapSimple').mined:
         store.dispatch(getAllBalancesForConnectedAddress())
         break
+      case 'CANCEL_SWAP':
+        makeMiddlewareEthersTransactionsFn(cancelSwap, 'cancelOrder', store, action, getSwapSimpleOrderId(action.order))
```

Comment:
Does this need to be` 'cancelSwap'` not `'cancelOrder'`?

Reply:
good catch

Reply:
fixed